bolt-modules/boltlib/lib/puppet/functions/run_command.rb

```diff
@@ -56,6 +57,23 @@ def run_command_with_description(command, targets, description = nil, options =
     options = options.transform_keys { |k| k.sub(/^_/, '').to_sym }
     options[:description] = description if description
 
+    # Ensure env_vars is a hash and that each hash value is transformed to JSON
+    # so we don't accidentally pass Ruby-style data to the target.
+    if options[:env_vars]
+      unless options[:env_vars].is_a?(Hash)
+        raise Bolt::ValidationError, "Option 'env_vars' must be a hash"
```

Comment:
Hmm in thinking more about it I think the right thing to do for this and the other validation errors here is to additionally validate them as part of the validate functions for the YAML steps in the 'step key' classes - that way they'll fail fast for YAML plans and include the line number there. It looks like that's what we do for other validation errors in plan functions.

Reply:
Updated to validate the type in the step itself. This doesn't include the line number, but does include the step name/number.